sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/AstBuilder.java

```diff
@@ -111,7 +112,16 @@ public UnresolvedPlan visitQuerySpecification(QuerySpecificationContext queryCon
       return project.attach(emptyValue);
     }
 
-    UnresolvedPlan result = project.attach(visit(queryContext.fromClause()));
+    // If limit (and offset) keyword exists:
+    // Add Limit node, plan structure becomes:
+    // Project -> Limit -> visit(fromClause)
+    // Else:
+    // Project -> visit(fromClause)
+    UnresolvedPlan from = visit(queryContext.fromClause());
+    if (queryContext.limitClause() != null) {
```

Comment:
Btw, so no Limit operator created if `LIMIT` clause absent in query. In this case what's the `size` in DSL generated? I recall we wanted to create a Limit operator with a default value here anyway?